### ISSUE TITLE
License clarification required

### ISSUE THREAD

**Issue by micolous** [acomminos/Plumble-Legacy#119]
According to the file LICENSE, the following conditions apply to Plumble:

> Copyright (C) 2010-2012, Benjamin Jemlich benjamin@jemlich.de
> 
> All rights reserved.

This appears to have been imported from the upstream project, @pcgod / mumble-android.

This would indicate that it is not an open source project (and shouldn't be distributed, modified, especially you shouldn't distribute your own binaries via Google Play store).  I also point direct you to this [post on the Mumble forum](http://sourceforge.net/apps/phpbb/mumble/viewtopic.php?p=3075#p3075):

> pcgod doesn't want to see that broken code spread so his github project isn't actually under an OSS license (and hence using it is technically not legal).
> 
> At this point in time I don't know of any core team efforts to implement an Android client. Sorry.
> 
> We would definitely like to see an Android client though so if anyone wants to work on that we'd gladly help out with stuff like protocol questions etc. (just visit us in IRC on #mumble @ freenode).

**Comment by acomminos** [comment]
Hello,

Thank you for bringing this problem to my attention. I will be in contact with the original creator of the app to see if we can resolve this issue.

**Comment by acomminos** [comment]
I have informed the original creator. I will keep this issue updated with any news.

**Comment by acomminos** [comment]
No response still. I sent another request to a different email, the one in the Mumble contributors dialog.

**Comment by Zuko** [comment]
go to IRC channel
9 cze 2013 06:47, "Andrew Comminos" notifications@github.com napisał(a):

**Comment by micolous** [comment]
It doesn't seem that @pcgod hasn't [made any code contributions in 9 months](https://github.com/pcgod?tab=activity).  It seems like he's stopped working on the Mumble project at all.  There may be more to this story, but I'd suggest talking on IRC with the other Mumble developers as they may know more.

There's also no mention of anyone successfully getting in touch with him about the license.  

Unfortunately that leaves you with the default position of "not allowed".

This would suggest that you need to pull Plumble from Google Play (and other distribution channels) and start a rewrite of the project.  From reading that forum, Mumble developers are pretty keen to have a working Android project.  That's your way forward from here.

You're in this position because:
- You've forked the repository at a point where it is explicitly stated "all rights reserved".
- Berne Convention says that everything is automatically copyrighted in the absence of notices stating otherwise.  This means even before the license statement was added, that **already** applied -- addition of the statement merely makes it explicit.  So you can't just fork from an older revision, or fork from one of the other forks to continue -- they all have exactly the same problem.
- There's statements on the Mumble project forums which indicate that @pcgod doesn't want people using his code (which begs the question "why did he release this booby-trapped code").

Although GitHub ToS requires that you allow viewing of source and forking.  However these are very limited rights... for example they don't explicitly allow distribution of forked binaries (Google Play).  This is more commonplace than you'd think -- Debian had to fork Firefox and change it's name in order to include their patches, as the Mozilla project refused to allow modified versions of Firefox to be distributed with their trademarks.

**Comment by acomminos** [comment]
I've tried to get in contact with him via IRC over the past while, to no avail. I'm going to do some diff trickery after I finish this audio rewrite to see if I can remove all of the original code and rewrite his code from scratch. According to the others in IRC, when asked about the app, he has opted to not take any stance whatsoever any more. I'd like to take advantage of this grey area in the mean time to do the rewrite.

Andrew

**Comment by micolous** [comment]
The problem is it's still a derivative work -- there's no "grey area" to it.

Stuff like `git blame` will only show you part of his code, as if you've reformatted that code, moved it around, or added parts to it, it's not going to identify as his.  You'll also need to identify where you've made a derivative of his code, because that needs to go **as well**.

I'm not overly familiar with the codebase, but how much of it is calling an API in the backend to talk to interface with the Mumble client?  It may be possible to just remove all of the code there, and then implement that API again yourself.  That way you can at least keep your UI code (provided that itself isn't a derivative), which seems like that is where you've done the majority of your work.

Unfortunately "grey" is still not allowed by copyright.  You need to get an explicit statement on the matter, and in the absence of such a statement, the default position is "not allowed".  If he doesn't want to take a stance on the matter (ie: not restrict the code), he needs to explicitly make a statement that he's releasing the code under a particular open source license, or releasing the code into the public domain.

**Comment by acomminos** [comment]
You're right in that 'grey' is definitely not a defence in copyright. The optimal solution, and the simplest, would be to get in contact with @pcgod directly to discuss the licensing concern. I'm attempting everything in my power to do this, as I've stated numerous times before.

The rewrite using git tools is obviously the worst-case scenario. Of course, I would still do screening; the main purpose of using tools such as blame would be to find what files I made myself and those that he contributed to. For example, files like `MumbleService.java` and `MumbleProtocol.java` are indeed derivative works and would need to be rewritten. I'm fully aware of this.

Lastly, with regards to the calling of an API, the app is very heavily integrated. The closest thing you'd be looking for would be the Android service in `MumbleService.java`. While Plumble has made vast improvements to the app's UI, features like Opus codec support as well as countless bug fixes and restructuring make it difficult to isolate a particular block of the code. If you check out the [contributors page](https://github.com/Morlunk/Plumble/contributors) you can see that I've rewrote many parts of the app entirely and am continuing to do so.

I will continue to keep this issue updated as I receive new info from other Mumble enthusiasts and developers. Thank you for your patience while I resolve this.

**Comment by MoriTanosuke** [comment]
I saw that you put the replacement `jumble` under Apache v2 at https://github.com/Morlunk/Jumble/blob/54eb77f638404a8c0ed0eb8b823f5644f3cbd1a1/LICENSE. Will Plumble get the same licensing after clarification of the current state?

**Comment by acomminos** [comment]
The "clean" version of Plumble I'm working on will be licensed under the GNU GPL v3. I'm a fan of strong copyleft, Jumble is only licensed under the Apache license so that proprietary apps can use the Mumble protocol in cool new ways (such as in games).
